tests/test_num_regression.py

```diff
@@ -160,7 +160,7 @@ def test_different_data_types(num_regression, no_regen):
     # Smoke test: Should not raise any exception
     num_regression.check({"data1": data1})
 
-    data2 = np.array(["a"] * 10)
+    data2 = np.array([True] * 10)
```

Comment:
python strings have 'object' dtype inside numpy/pandas. Objects are no longer accepted since we can not test it numerically (or can we? Maybe I'm missing something).
But boolean type is numerical-like, then the dataframe is not rejected and the check returns the expected error message

Reply:
Ah, okay, but perhaps we could extend it to support simple "value-type" objects such as strings, for an improved usage and for making user lives easier? They would not compare numerically, just for exact matches in these cases. (Could be another PR/issue though)

Reply:
Did it! Rolled back the `num_regression` test that I changed to use string arrays again

Reply:
Awesome!